src/repositories/repoBranchesRepository.ts

```diff
@@ -8,13 +8,44 @@ export class RepoBranchesRepository extends BaseRepository {
     super(config, logger, 'RepoBranchesRepository', db.collection(config.get('repoBranchesCollection')));
   }
 
+  async getConfiguredBranchesByGithubRepoName(repoName: string): Promise<any> {
+    const query = { repoName: repoName };
+    const reposObject = await this.findOne(
+      query,
+      `Mongo Timeout Error: Timedout while retrieving repos entry for ${repoName}`
+    );
+    if (reposObject?.branches) {
+      return {
+        branches: reposObject.branches,
+        repoName: reposObject.repoName,
+        status: 'success',
+      };
+    } else {
+      return { status: 'failure' };
+    }
+  }
+
   async getProjectByRepoName(repoName: string) {
     const query = { repoName };
     const projection = { _id: 0, project: 1 };
     const res = await this.findOne(query, `Error while getting project by repo name ${repoName}`, { projection });
     return res.project;
   }
 
+  async getRepo(repoName: string): Promise<any> {
+    const query = { repoName: repoName };
+    const repo = await this.findOne(query, `Mongo Timeout Error: Timedout while retrieving branches for ${repoName}`);
+    // if user has specific entitlements
+    return repo;
+  }
+
+  async getRepoBranches(repoName: string): Promise<any> {
```

Comment:
With these changes, are we able to safely delete the `branchRepository.ts` file?

Reply:
@rayangler yes, but I just wanted to make sure we were good with the testing, which we are, and I wanted to make sure the feedback was good as well. My plan was to remove it before merging in the code.

Reply:
Should be good to remove 👍